pkg/operator/ceph/object/realm/controller.go

```diff
@@ -176,6 +198,31 @@ func (r *ReconcileObjectRealm) reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileObjectRealm) pullCephRealm(realm *cephv1.CephObjectRealm) (reconcile.Result, error) {
+	realmArg := fmt.Sprintf("--rgw-realm=%s", realm.Name)
+	urlArg := fmt.Sprintf("--url=%s", realm.Spec.Pull.Endpoint)
+	logger.Debug("getting keys to pull realm")
+	accessKeyArg, secretKeyArg, err := object.GetRealmKeyArgs(r.context, realm.Name, realm.Namespace)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
```

Comment:
@travisn am I doing the correct type of error checking here for getting the secrets? We want to keep re-queuing until the secret is found right?

Reply:
correct, we want to requeue if not found